### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by tobuya**
- Create a class `Classroom` with the `@label` instance variable, which should be initialized in the constructor. Setter and getter for `@label`.
- Create the `has-many/belongs-to` relationship between `Classroom` and `Student`: Create the has-many side (a classroom has many students), and Create the belongs-to side (a student belongs to a classroom), Make sure that when adding a student to a classroom it also sets the classroom for the student, Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class `Book` with the `@title` and `@author` instance variables, which should be initialized in the constructor. Setters and getters for instance variables
- Create a class Rental with the `@date` instance variable, which should be initialized in the constructor. Setter and getter for `@date`
- Create the `many-to-many` (also has-many-through) relationship between `Person` and `Book` using the intermediate class `Rental`.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
   def initialize(age, name, parent_permission: true)
     super()
     @age = age
     @name = name
     @parent_permission = parent_permission
+    @rentals = []
   end
 
-  attr_reader :id
-  attr_accessor :name, :age
```
Comment (Wes-Isaac) [inline person.rb:12]: - [ ] Great job adding rentals in the constructor, here we also need a method (add_rental) to add rentals, the person itself should be sent as a parameter to create the new rental achieving the 'has-many' association.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_reader :rentals
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
```
Comment (Wes-Isaac) [inline book.rb:8]: - [ ] Great job adding rentals in the constructor, here we also need a method (add_rental) to add rentals, the person itself should be sent as a parameter to create the new rental achieving the 'has-many' association.
`Rental.new(...)`

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_reader :rentals
```
Comment (Wes-Isaac) [inline book.rb]: - [ ] Great job, here we need both setters and getters for rentals. Kindly implement that.

File: rental.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+class Rental
+  attr_reader :book, :person
```
Comment (Wes-Isaac) [inline rental.rb]: - [ ] Great job, here we need both setters and getters for `@book `and `@person `instance variables. Kindly implement that

File: person.rb [hunk 5]
```diff
@@ -1,16 +1,17 @@
 require './nameable'
 
 class Person < Nameable
+  attr_reader :id, :rentals
+  attr_accessor :name, :age
```
Comment (Wes-Isaac) [inline person.rb]: - [ ] Great job, here we need both setters and getters for rentals. Kindly implement that.

**Review by Wes-Isaac — CHANGES_REQUESTED** [review]
Hi @tobuya,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Professionally documented your work 💯.
- Linters are all green ✅.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: person.rb (hunk 5 above)
Comment (tobuya) [reply person.rb]: @Wes-Isaac Thank you very much for all the suggestions.

**Review by prolajumokeoni — APPROVED** [review]
Hi @tobuya 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- All changes requested have been implemented   ✅ 


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**